runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1073,29 +1073,11 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 		J9IndexableObject *j9ArrayPtr = (J9IndexableObject *)arrayPtr;
 
 		if (InlineContiguous == getPreservedArrayLayout(forwardedHeader)) {
-			setDataAddrForContiguous(j9ArrayPtr);
-		} else {
-			setDataAddrForDiscontiguous(j9ArrayPtr, NULL);
-		}
-#endif /* J9VM_ENV_DATA64 */
-	}
-	/**
-	 * External fixup dataAddr API to update pointer of indexable objects.
-	 * Sets the dataAddr of either a contiguous or discomtiguous indexable
-	 * object.
-	 *
-	 * @param arrayPtr      Pointer to the indexable object whose size is required
-	 */
-	MMINLINE void
-	fixupDataAddr(omrobjectptr_t arrayPtr)
-	{
-#if defined(J9VM_ENV_DATA64)
-		J9IndexableObject *j9ArrayPtr = (J9IndexableObject *)arrayPtr;
-
-		if (InlineContiguous == getArrayLayout(j9ArrayPtr)) {
-			setDataAddrForContiguous(j9ArrayPtr);
+			if (shouldFixupDataAddrForContiguous((J9IndexableObject *)forwardedHeader->getObject())) {
+				setDataAddrForContiguous(j9ArrayPtr);
+			}
```

Comment:
perhaps we can put a comment what can occur in else clause (since I don't think we can add a cheap assert). if I understand 2 cases are:
- running with offheap/double mapping, and data is offheap 
- running with arraylets 

interestingly if we leave dataAddr uninitialized for arraylet case we can even end up fixing it, if it jsut happen to point to  end of header. unlikely, but possible and harmless

Reply:
meanwhile I realized I was incorrect for arraylet case. dataAddr will be set and needs to be fixed up in that case.
this is because array access macro for contiguous just checks if dataAddr is present (and if it is present, it assumes it's initialized)

Reply:
Based on [this](https://github.com/eclipse-openj9/openj9/pull/14667#discussion_r1185479490) comment, is a comment in the else clause of `fixupDataAddr` still needed to provide additional context?